include/VectorGraph.h

```diff
@@ -0,0 +1,739 @@
+	//float processLineTypeRandom(float xIn, float valA, float valB, float curve, float fadeInStartLoc);
+	std::vector<float> processLineTypeArrayRandom(std::vector<float>* xArray, unsigned int startLoc, unsigned int endLoc,
+		float valA, float valB, float curve, float fadeInStartLoc);
+
+	// updating
+	// adds the points that are
+	// effected by the effector's values changing
+	// ONLY WORKS IN SORTED ARRAYS
+	void getUpdatingFromEffector(std::vector<unsigned int>* updatingPointLocations);
+	// if pointLocation >= 0 -> adds the location to m_needsUpdating
+	// else it will update the whole m_dataArray and m_bakedSamples
+	// changes in the size of m_dataArray (addtition, deletion, ect.)
+	// should to cause a full update
+	void getUpdatingFromPoint(int pointLocation);
+	// adds the points that are changed because their
+	// automation is changed
+	void getUpdatingFromAutomation();
+	// recalculates and sorts m_needsUpdating so
+	// every point is in there only once
+	void getUpdatingOriginals();
+
+	// real getSamples processing
+	void getSamples(unsigned int targetSizeIn, bool* isChangedOut, std::vector<unsigned int>* updatingValuesOut, std::vector<float>* sampleBufferOut);
+	// gets every m_needsUpdating point's line's start and end effector point's location in the effector dataArray
+	// .first = start, .second = line end location (effector dataArray)
+	//void getSamplesLocations(VectorGraphDataArray* effector, std::vector<std::pair<unsigned int, unsigned int>>* effectorDataOut);
+	void getSamplesUpdateLines(VectorGraphDataArray* effector, std::vector<float>* effectorSamples,
+		std::vector<float>* outputXLocations, unsigned int iIn, float stepSize);
+	bool isEffectedPoint(unsigned int pointLocation);
+
+	// checks m_isFixedEndPoints, does not call dataChanged()
+	void formatDataArrayEndPoints();
+
+	// can new data be added or removed
+	bool m_isFixedSize;
+	// can the positions be changed
+	bool m_isFixedX;
+	// can the values be changed
+	bool m_isFixedY;
+	// if true then it makes the last position coordinate 1, 1, the first point coordinate to -1 (ot 0), 0
+	bool m_isFixedEndPoints;
+	// can VectorGraphView select this
+	bool m_isSelectable;
+	// can the point attributes be edited
+	// every attribute outside of x and y
+	// automation can be changed
+	bool m_isEditableAttrib;
+	// can the points be automated or effected
+	// (can these settings be changed)
+	bool m_isAutomatableEffectable;
+	// if VectorGraphDataArray is allowed to save this
+	bool m_isSaveable;
+	// can values be less than 0
+	bool m_isNonNegative;
+
+	QColor m_lineColor;
+	QColor m_activeColor;
+	QColor m_fillColor;
+	QColor m_automatedColor;
+
+	VectorGraphModel* m_parent;
+	// simple id system for setEffectorArrayLocation
+	int m_id;
+
+	// which VectorGraphDataArray can effect this one, -1 if not effected
+	int m_effectorLocation;
+
+	// ordered array of VectorGraphPoints
+	std::vector<VectorGraphPoint> m_dataArray;
+	
+
+	// baking
```

Comment:
Cookies? :)

Reply:
Got the name from texture baking. I guess what you are trying to say is this name does not fit in...

Reply:
Oh, no. I was just leaving some fun commentary for a laugh. It's a big PR and going through a ton of lines can get mind numbing.
If the phase name makes sense, then it's fine to keep.

Reply:
Ok no problem. I appreciate your commentary. This is a big pr and even as someone who understands it reviewing all the comments and functions was tiring. Thanks for reviewing this.